Singer.API/Services/EmailService.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Net;
+using System.Net.Mail;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Singer.Data.Models.Configuration;
+using Singer.DTOs.Users;
+using Singer.Services.Interfaces;
+
+namespace Singer.Services
+{
+   public class EmailService<TUserDTO> : IEmailService<TUserDTO>
+   where TUserDTO : IUserDTO
+   {
+      private readonly SmtpClient _smtp;
+      private readonly string _mailFrom;
+
+      public EmailService(IOptions<EmailOptions> options = null)
+      {
+         if(options == null)
+         {
+            throw new ArgumentNullException(nameof(options));
+         }
+         var emailOptions = options.Value;
+         _smtp = new SmtpClient
+         {
+            Credentials = new NetworkCredential(emailOptions.SmtpUsername, emailOptions.SmtpPassword),
+            Host = emailOptions.SmtpHost,
+            Port = emailOptions.SmtpPort
+         };
+         _mailFrom = emailOptions.MailFrom;
+      }
+      public async Task SendAccountDetailsAsync(TUserDTO user, string resetPasswordLink)
+      {
+         var msg = new MailMessage(_mailFrom, user.Email);
```

Comment:
It would be slightly nicer to put the email in html format (using `msg.IsBodyHtml = true;` and putting some anchor tags in the body with the hyperlink to the reset page. Currently it depends on the email client whether the link will be clickable or not